CONTRIBUTING.md

```diff
@@ -27,6 +27,31 @@ into the following categories:
    - Pick an issue and comment on the task that you want to work on this feature.
    - If you need more context on a particular issue, please ask and we shall provide.
 
+<!-- ############################################################################################################### -->
+
+## Table of Contents
+- [Table of Contents](#table-of-contents)
+- [Developing Ignite](#developing-ignite)
+   - [Quickstart guide for first-time contributors](#quickstart-guide-for-first-time-contributors)
```

Comment:
I think using 2 spaces for nesting looks much better and follows common md conventions.
Take a look at https://github.com/pytorch/ignite/blob/master/CONTRIBUTING.md#codebase-structure for reference

```suggestion
  - [Quickstart guide for first-time contributors](#quickstart-guide-for-first-time-contributors)
  
  the same for other items.
```